src/app/upgrade.h

```diff
@@ -106,26 +118,62 @@ bool upgradeResumeFile(const QString &filepath)
 
 bool upgrade(bool ask = true)
 {
+    QIniSettings *oldResumeSettings = new QIniSettings("qBittorrent", "qBittorrent-resume");
+    QString oldResumeFilename = oldResumeSettings->fileName();
+    QVariantHash oldResumeData = oldResumeSettings->value("torrents").toHash();
+    delete oldResumeSettings;
+    if (oldResumeData.isEmpty()) {
+        Utils::Fs::forceRemove(oldResumeFilename);
+        return true;
+    }
+
+    if (ask && !userAcceptsUpgrade()) return false;
+
+    int maxPrio = 0;
     QString backupFolderPath = Utils::Fs::expandPathAbs(Utils::Fs::QDesktopServicesDataLocation() + "BT_backup");
     QDir backupFolderDir(backupFolderPath);
-    if (!backupFolderDir.exists()) return true;
-
     QStringList backupFiles = backupFolderDir.entryList(QStringList() << QLatin1String("*.fastresume"), QDir::Files, QDir::Unsorted);
-    if (!backupFiles.isEmpty()) {
-        if (ask && !userAcceptsUpgrade()) return false;
-
-        QRegExp rx(QLatin1String("^([A-Fa-f0-9]{40})\\.fastresume$"));
-        foreach (QString backupFile, backupFiles) {
-            if (rx.indexIn(backupFile) != -1) {
-                if (!upgradeResumeFile(backupFolderDir.absoluteFilePath(backupFile)))
-                    Logger::instance()->addMessage(QObject::tr("Couldn't migrate torrent with hash: %1").arg(rx.cap(1)), Log::WARNING);
-            }
-            else {
-                Logger::instance()->addMessage(QObject::tr("Couldn't migrate torrent. Invalid fastresume file name: %1").arg(backupFile), Log::WARNING);
-            }
+    QRegExp rx(QLatin1String("^([A-Fa-f0-9]{40})\\.fastresume$"));
+    foreach (QString backupFile, backupFiles) {
+        if (rx.indexIn(backupFile) != -1) {
+            if (upgradeResumeFile(backupFolderDir.absoluteFilePath(backupFile), oldResumeData[rx.cap(1)].toHash(), maxPrio))
+                oldResumeData.remove(rx.cap(1));
+            else
+                Logger::instance()->addMessage(QObject::tr("Couldn't migrate torrent with hash: %1").arg(rx.cap(1)), Log::WARNING);
+        }
+        else {
+            Logger::instance()->addMessage(QObject::tr("Couldn't migrate torrent. Invalid fastresume file name: %1").arg(backupFile), Log::WARNING);
+        }
+    }
+
+    foreach (const QString &hash, oldResumeData.keys()) {
+        QVariantHash oldTorrent = oldResumeData[hash].toHash();
+        if (oldTorrent.value("is_magnet", false).toBool()) {
+            libtorrent::entry resumeData;
+            resumeData["qBt-magnetUri"] = Utils::String::toStdString(oldTorrent.value("magnet_uri").toString());
+            resumeData["qBt-paused"] = true;
```

Comment:
Why `true`?
IIRC, before v3.2.x qbt never remembered the paused state of undownloaded magnets. This should remain false.
I prefer you make this `false`